wix-embedded-mysql/src/test/scala/com/wix/mysql/EmbeddedMysqlTest.scala

```diff
@@ -0,0 +1,85 @@
+package com.wix.mysql
+
+import com.wix.mysql.EmbeddedMysql._
+import com.wix.mysql.config.Charset
+import com.wix.mysql.config.Charset.{aCharset, LATIN1, UTF8MB4}
+import com.wix.mysql.config.MysqldConfig.aMysqldConfig
+import com.wix.mysql.config.SchemaConfig.aSchemaConfig
+import com.wix.mysql.distribution.Version.v5_6_latest
+
+/**
+ * @author viliusl
+ * @since 03/07/15
+ */
+class EmbeddedMysqlTest extends IntegrationTest {
+
+  "EmbeddedMysql instance" should {
+
+    "start with default values" in {
+      val config = aMysqldConfig(v5_6_latest).build
+
+      mysqld = anEmbeddedMysql(config).start
+
+      serverCharset mustEqual UTF8MB4
+      userCanConnect("auser", "sa", 3310) must beTrue
+    }
+
+    "use custom values provided via MysqldConfig" in {
+      val config = aMysqldConfig(v5_6_latest)
+        .withCharset(LATIN1)
+        .withUser("zeUser", "zePassword")
+        .withPort(1112)
+        .build
+
+      mysqld = anEmbeddedMysql(config).start
+
+      serverCharset mustEqual LATIN1
+      userCanConnect("zeUser", "zePassword", 1112) must beTrue
+    }
+  }
+
+  "EmbeddedMysql schemas" should {
+    "use defaults" in {
+      val config = aMysqldConfig(v5_6_latest).build
+
+      mysqld = anEmbeddedMysql(config)
+        .addSchema("aSchema")
+        .start
+
+      schemaCharset("aSchema") mustEqual UTF8MB4
+      userCanConnect("auser", "sa", 3310, "aSchema") must beTrue
+    }
+
+    "use custom values" in {
+      val config = aMysqldConfig(v5_6_latest).build
+      val schema = aSchemaConfig("aSchema")
+        .withCharset(LATIN1)
+        .build
+
+      mysqld = anEmbeddedMysql(config)
+        .addSchema(schema)
+        .start
+
+      schemaCharset("aSchema") mustEqual LATIN1
+      userCanConnect("auser", "sa", 3310, "aSchema") must beTrue
+    }
+  }
+
```

Comment:
we need to extract all matchers out of here and clean this test up, I would probably prefer matchers like this:

```scala
  def haveCharsetOf(charset: Charset, onSchema: String): Matcher[EmbeddedMysql] =
    beTypedEqualTo(charset) ^^ {
      mySql: EmbeddedMysql =>
        val ds = aDataSource(mysqld.getConfig, onSchema)
        aCharset(
          aSelect[String](ds, sql = s"SELECT default_character_set_name FROM information_schema.SCHEMATA where SCHEMA_NAME = '$onSchema';"),
          aSelect[String](ds, sql = s"SELECT DEFAULT_COLLATION_NAME FROM information_schema.SCHEMATA where SCHEMA_NAME = '$onSchema';"))
    }

  def beAvailableOn(port: Int, withUser: String, password: String, andSchema: String = "information_schema"): Matcher[EmbeddedMysql] =
   beTrue ^^ { mySql: EmbeddedMysql => aSelect[java.lang.Long](aDataSource(withUser, password, port, andSchema), "select 1;") == 1 }
```